addons/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/blob/s3/S3BlobStore.java

```diff
@@ -0,0 +1,782 @@
+            t0 = System.currentTimeMillis();
+            log.debug("Copying s3://" + sourceBucketName + "/" + sourceBucketKey + " to s3://" + bucketName + "/"
+                    + bucketKey);
+        }
+
+        if (getKeyStrategy().useDeDuplication() && exists(bucketKey)) {
+            return true;
+        }
+
+        // copy the blob
+        logTrace("->", "getObjectMetadata");
+        ObjectMetadata sourceMetadata = amazonS3.getObjectMetadata(sourceBucketName, sourceBucketKey);
+        // don't catch AmazonServiceException if missing, caller will do it
+        long length = sourceMetadata.getContentLength();
+        logTrace("<-", length + " bytes");
+        try {
+
+            copyBlob(sourceBlobStore.config, sourceBucketKey, config, bucketKey, move);
+
+            if (log.isDebugEnabled()) {
+                long dtms = System.currentTimeMillis() - t0;
+                log.debug("Copied s3://" + sourceBucketName + "/" + sourceBucketKey + " to s3://" + bucketName + "/"
+                        + bucketKey + " in " + dtms + "ms");
+            }
+            return true;
+        } catch (AmazonServiceException e) {
+            logTrace("<--", "ERROR");
+            String message = "Direct copy failed from s3://" + sourceBucketName + "/" + sourceBucketKey + " to s3://"
+                    + bucketName + "/" + bucketKey + " (" + length + " bytes)";
+            log.warn(message + ", falling back to slow copy: " + e.getMessage());
+            log.debug(message, e);
+            return false;
+        }
+    }
+
+    protected void copyBlob(S3BlobStoreConfiguration sourceConfig, String sourceKey,
+            S3BlobStoreConfiguration destinationConfig, String destinationKey, boolean move) {
+        CopyObjectRequest copyObjectRequest = new CopyObjectRequest(sourceConfig.bucketName, sourceKey,
+                destinationConfig.bucketName, destinationKey);
+        if (destinationConfig.useServerSideEncryption) {
+            // server-side encryption
+            if (isNotBlank(destinationConfig.serverSideKMSKeyID)) {
+                // SSE-KMS
+                SSEAwsKeyManagementParams params = new SSEAwsKeyManagementParams(destinationConfig.serverSideKMSKeyID);
+                copyObjectRequest.setSSEAwsKeyManagementParams(params);
+            } else {
+                // SSE-S3
+                ObjectMetadata newObjectMetadata = new ObjectMetadata();
+                newObjectMetadata.setSSEAlgorithm(ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION);
+                // CopyCallable.populateMetadataWithEncryptionParams will get the rest from the source
+                copyObjectRequest.setNewObjectMetadata(newObjectMetadata);
+            }
+            // TODO SSE-C
+        }
+        logTrace("->", "copyObject");
+        logTrace("hnote right: " + sourceKey + " to " + destinationKey);
+        Copy copy = destinationConfig.transferManager.copy(copyObjectRequest, sourceConfig.amazonS3, null);
```

Comment:
Is the `copy` method that will be responsible of moving from `Standard` S3 storage into `Glacier`.  if it is true i don't see the how it will be done, i was looking for a call on `com.amazonaws.services.s3.model.CopyObjectRequest#setStorageClass(com.amazonaws.services.s3.model.StorageClass)`.

i looked also to `org.nuxeo.ecm.blob.s3.S3BlobStore#writeBlobProperties` and i didn't found a call to `com.amazonaws.services.s3.AmazonS3#changeObjectStorageClass`

Reply:
Yes we'll end up calling `changeObjectStorageClass` from `writeBlobProperties` but this code is not written yet.